core/src/main/java/org/apache/iceberg/actions/RewriteDataFilesCommitManager.java

```diff
@@ -51,18 +53,25 @@ public RewriteDataFilesCommitManager(Table table, long startingSnapshotId) {
 
   public RewriteDataFilesCommitManager(
       Table table, long startingSnapshotId, boolean useStartingSequenceNumber) {
-    this(table, startingSnapshotId, useStartingSequenceNumber, ImmutableMap.of());
+    this(
+        table,
+        startingSnapshotId,
+        useStartingSequenceNumber,
+        ImmutableMap.of(),
+        SnapshotRef.MAIN_BRANCH);
   }
 
   public RewriteDataFilesCommitManager(
       Table table,
       long startingSnapshotId,
       boolean useStartingSequenceNumber,
-      Map<String, String> snapshotProperties) {
+      Map<String, String> snapshotProperties,
+      String branch) {
```

Comment:
did revapi not complain about breaking this interface?  Should we keep the other for backward compat?